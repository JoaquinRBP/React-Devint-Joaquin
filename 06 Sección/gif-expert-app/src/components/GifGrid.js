import React from 'react';
import { GifGridItem } from './GifGridItem';
import {useFetchGifs} from '../hooks/useFetchGifs';
//import React, {useState, useEffect} from 'react';
//import {getGifs} from './../helpers/getGifs';
export const GifGrid = ({category})=>{
    //const [Count,setCount] = useState(0);
    // const [images, setimages] = useState([])
    // useEffect(()=>{
    //     getGifs(category)
    //     .then(setimages)
    // }, [category]) //segundo arreglo es una lista de dependencias
    const {data: imagenes,loading} = useFetchGifs(category)
    
    return (
        <>
        <h3>{category}</h3>
        {loading && <p className="animate__animated animate__flash">Loading</p>}
        <div className="card-grid">
            <ol>{
                imagenes.map(img =>(
                    <GifGridItem 
                    key={img.id}
                    //img={img}
                    {...img}
                    />
                ))}
            </ol>
        </div>
        </>
    )
}