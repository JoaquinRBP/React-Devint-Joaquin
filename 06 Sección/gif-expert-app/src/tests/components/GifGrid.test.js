import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import {useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');
describe('Pruebas sobre GifGrid',()=>{
    const category = 'Pruebitas';
    useFetchGifs.mockReturnValue({
        data: [],
        loading: true
    });//regresar el valor asignado
    const wrapper = shallow(<GifGrid category={category} />);
    test('GifGrid snapshoot',()=>{
        expect(wrapper).toMatchSnapshot();
    });
    test('Debe de mostrar items cuando se cargan imagenes',()=>{
        const gifs = [{
            id:'ABC',
            url:'https://localhost/cualquier/cosa.jpg',
            title:'Cualquier cosa'
        }];
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });//regresar el valor asignado
        const wrapper = shallow(<GifGrid category={category} />);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(1);
    })
});