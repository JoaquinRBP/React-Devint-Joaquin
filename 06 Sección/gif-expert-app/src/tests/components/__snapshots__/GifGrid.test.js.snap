// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre GifGrid Debe de mostrar items cuando se cargan imagenes 1`] = `
<Fragment>
  <h3>
    Pruebitas
  </h3>
  <div
    className="card-grid"
  >
    <ol>
      <GifGridItem
        id="ABC"
        key="ABC"
        title="Cualquier cosa"
        url="https://localhost/cualquier/cosa.jpg"
      />
    </ol>
  </div>
</Fragment>
`;

exports[`Pruebas sobre GifGrid GifGrid snapshoot 1`] = `
<Fragment>
  <h3>
    Pruebitas
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  >
    <ol />
  </div>
</Fragment>
`;
